datastore/client.go

```diff
@@ -42,11 +42,15 @@ type datastoreClient struct {
 	md metadata.MD
 }
 
-func newDatastoreClient(conn grpc.ClientConnInterface, projectID string) pb.DatastoreClient {
+func newDatastoreClient(conn grpc.ClientConnInterface, projectID, databaseID string) pb.DatastoreClient {
+	resourcePrefixValue := "projects/" + projectID
+	if databaseID != "" {
```

Comment:
maybe add a nil check here as well?

Reply:
databaseID is of type string. In Go, strings cannot be nil.
`if databaseID != nil && databaseID != ""` throws error 

`invalid operation: databaseID != nil (mismatched types string and untyped nil)`